toolchain/check/eval.cpp

```diff
@@ -1311,6 +1324,43 @@ static auto MakeConstantForBuiltinCall(Context& context, SemIRLoc loc,
       return SemIR::ConstantId::NotConstant;
     }
 
+    case SemIR::BuiltinFunctionKind::TypeAnd: {
+      CARBON_CHECK(arg_ids.size() == 2);
+      auto lhs_facet_type_id = SemIR::FacetTypeId::None;
+      auto rhs_facet_type_id = SemIR::FacetTypeId::None;
+      for (auto [facet_type_id, arg_id] :
+           llvm::zip(std::to_array({&lhs_facet_type_id, &rhs_facet_type_id}),
+                     arg_ids)) {
+        if (auto facet_type =
+                context.insts().TryGetAs<SemIR::FacetType>(arg_id)) {
+          *facet_type_id = facet_type->facet_type_id;
+        } else {
+          CARBON_DIAGNOSTIC(
+              FacetTypeRequiredForTypeAndOperator, Error,
+              "BitAnd operator on types requires facet type on both sides");
+          // TODO: Find a location for the lhs or rhs specifically, instead of
+          // the whole thing. If that's not possible we can change the text to
+          // say if it's referring to the left or the right side for the error.
+          context.emitter().Emit(loc, FacetTypeRequiredForTypeAndOperator);
```

Comment:
Instead of `loc`, can you pass `arg_id` here?

Reply:
FTR this is weird to me, I think the instruction should have a location. But that's not for this PR.